rellic/AST/Util.cpp

```diff
@@ -68,6 +68,45 @@ bool ReplaceChildren(clang::Stmt *stmt, StmtMap &repl_map) {
   return change;
 }
 
+clang::QualType GetLeastIntTypeForBitWidth(clang::ASTContext &ctx,
+                                           unsigned size, unsigned sign) {
+  auto &ti{ctx.getTargetInfo()};
+  auto target_type{ti.getLeastIntTypeByWidth(size, sign)};
+  return ctx.getIntTypeForBitwidth(ti.getTypeWidth(target_type), sign);
+}
+
+clang::Expr *CastExpr(clang::ASTContext &ctx, clang::QualType dst,
+                      clang::Expr *op) {
+  // Get operand type
+  auto src = op->getType();
+  // Helpers
+  auto IsInt = [&ctx](clang::QualType t) { return t->isIntegralType(ctx); };
+  auto IsFloat = [](clang::QualType t) { return t->isFloatingType(); };
+  auto IsSmaller = ctx.getTypeSize(src) < ctx.getTypeSize(dst);
+  auto MakeCast = [&ctx, dst, op](clang::CastKind kind) {
+    return CreateCStyleCastExpr(ctx, dst, kind, op);
+  };
+
+  // CK_FloatingCast
```

Comment:
Can you change these comments to describe operation being detected/handled? E.g, float to float and issmaller could be described as "Widen the floating point type, e.g. `fpext` in LLVM bitcode."

What happens if source and dest operand types are different, but the size is the same? e.g. `int64_t` to `double`, or `int32_t` to `float`?

Reply:
>What happens if source and dest operand types are different, but the size is the same? e.g. int64_t to double, or int32_t to float?

This should be covered by the latter two cases. Just a simple `IntegralToFloating` cast.